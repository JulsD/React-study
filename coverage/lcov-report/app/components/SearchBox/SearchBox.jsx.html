<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for app/components/SearchBox/SearchBox.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">app/components/SearchBox</a> SearchBox.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">8.33% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>1/12</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">8.33% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>1/12</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import { withRouter } from 'react-router-dom';
import { Field, reduxForm, SubmissionError } from 'redux-form'
&nbsp;
import Button from '../Button';
import styles from './search-box.css';
&nbsp;
const renderRadioField = <span class="fstat-no" title="function not covered" >({</span>label, input}) =&gt; (
<span class="cstat-no" title="statement not covered" >  &lt;Button type='label' kind={`${input.checked ? 'primary' : 'not-active'}`} className={styles.switcherBtn}&gt;</span>
    {label}
    &lt;input type='radio' {...input}/&gt;
  &lt;/Button&gt;
)
&nbsp;
class SearchBox extends React.Component {
<span class="fstat-no" title="function not covered" >  co</span>nstructor(props) {
<span class="cstat-no" title="statement not covered" >    super(props);</span>
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  su</span>bmit({searchText=<span class="branch-0 cbranch-no" title="branch not covered" >'',</span> searchBy=<span class="branch-0 cbranch-no" title="branch not covered" >''}</span>) {
&nbsp;
    let error = <span class="cstat-no" title="statement not covered" >{};</span>
    let isError = <span class="cstat-no" title="statement not covered" >false;</span>
&nbsp;
<span class="cstat-no" title="statement not covered" >    if(searchText.trim() === '' || searchBy.trim() === '') {</span>
<span class="cstat-no" title="statement not covered" >      isError = true;</span>
    }
&nbsp;
<span class="cstat-no" title="statement not covered" >    if (isError) {</span>
<span class="cstat-no" title="statement not covered" >      throw new SubmissionError(error);</span>
    } else {
<span class="cstat-no" title="statement not covered" >      this.props.history.push({</span>
        pathname: '/search',
        search: searchBy + '=' + searchText
      })
    }
&nbsp;
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  re</span>nder() {
    const { handleSubmit } = <span class="cstat-no" title="statement not covered" >this.props;</span>
<span class="cstat-no" title="statement not covered" >    return (</span>
      &lt;form className = {styles.root} onSubmit={handleSubmit(this.submit.bind(this))}&gt;
        &lt;label&gt;
          &lt;h3 className = {styles.title}&gt;Find your movie&lt;/h3&gt;
          &lt;Field name='searchText'
                 component='input'
                 className = {styles.searchInput}
                 type='search'
                 placeholder='Search movie'/&gt;
        &lt;/ label&gt;
        &lt;div className = {styles.switcher}&gt;
          &lt;h4 className = {styles.switcherTitle}&gt;Search by&lt;/h4&gt;
          &lt;Field type='radio'
                 name='searchBy'
                 value='title'
                 component={renderRadioField}
                 label='Title'/&gt;
          &lt;Field type='radio'
                 component={renderRadioField}
                 name='searchBy'
                 value='director'
                 label='Director'/&gt;
        &lt;/div&gt;
        &lt;div className = {styles.submit}&gt;
          &lt;Button type='submit' kind='primary'&gt;Search&lt;/Button&gt;
        &lt;/div&gt;
      &lt;/form&gt;
    );
  }
}
&nbsp;
export default reduxForm({
    form: 'search',
    initialValues: {
      searchText: '',
      searchBy: 'title'
    }
})(withRouter(SearchBox));</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Nov 09 2017 00:12:39 GMT+0100 (CET)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
