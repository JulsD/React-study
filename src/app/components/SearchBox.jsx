import React from 'react';

import btn from './../../styles/buttons.css';
import styles from './../../styles/search-box.css';

class SearchBox extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      searchText: '',
      searchBy: 'title'
    };

    this.handleInputChange = this.handleInputChange.bind(this);
  }

  handleInputChange(event) {
    const target = event.target;
    const value = target.value;
    const name = target.name;

    console.log(name + ':' + value);

    this.setState({
      [name]: value
    });
  }

  render() {
    return (
        <form className = {styles.root}>
          <label>
            <h3 className = {styles.title}>Find your movie</h3>
            <input name="searchText"
                   className = {styles.searchInput}
                   type="search"
                   placeholder="Search movie"
                   value={this.state.searchText}
                   onChange={this.handleInputChange}/>
          </ label>
          <div className = {styles.switcher}>
            <h4 className = {styles.switcherTitle}>Search by</h4>
            <label className={`${btn.btn} ${this.state.searchBy ==="title" ? btn.btnRed : btn.btnNotActive} ${styles.switcherBtn}`}>
              Title
              <input type="radio"
                     name="searchBy"
                     value="title"
                     onChange={this.handleInputChange}
                     checked={this.state.searchBy ==="title"}/>
            </label>
            <label className={`${btn.btn} ${this.state.searchBy ==="director" ? btn.btnRed : btn.btnNotActive} ${styles.switcherBtn}`}>
              Director
              <input type="radio"
                     name="searchBy"
                     value="director"
                     onChange={this.handleInputChange}
                     checked={this.state.searchBy ==="director"} />
            </label>
          </div>
          <div className = {styles.submit}><button type="submit" className = {`${btn.btn} ${btn.btnRed}`}>Search</button></div>
        </form>
    );
  }

}

export default SearchBox;
