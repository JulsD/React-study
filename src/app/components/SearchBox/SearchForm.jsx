import React from 'react';
import { Field, reduxForm } from 'redux-form'

import Button from '../Button';
import styles from './search-box.css';

const Form = props => {
  const { handleSubmit } = props
  return (
      <form className = {styles.root} onSubmit={handleSubmit}>
        <label>
          <h3 className = {styles.title}>Find your movie</h3>
          <Field name='searchText'
                 component='input'
                 className = {styles.searchInput}
                 type='search'
                 placeholder='Search movie'
                 onChange={handleInputChange}/>
        </ label>
        <div className = {styles.switcher}>
          <h4 className = {styles.switcherTitle}>Search by</h4>
          <Button type='label' kind={`${this.searchBy ==='title' ? 'primary' : 'not-active'}`} className={styles.switcherBtn}>
            Title
            <Field type='radio'
                   component='input'
                   name='searchBy'
                   value='title'
                   onChange={handleInputChange}
                   checked={this.searchBy ==='title'}/>
          </Button>
          <Button type='label' kind={`${this.searchBy ==='director' ? 'primary' : 'not-active'}`} className={styles.switcherBtn}>
            Director
            <Field type='radio'
                   component='input'
                   name='searchBy'
                   value='director'
                   onChange={handleInputChange}
                   checked={this.searchBy ==='director'} />
          </Button>
        </div>
        <div className = {styles.submit}>
          <Button type='submit' kind='primary'>Search</Button>
        </div>
      </form>
  );
}

const SearchForm = reduxForm({
    form: 'searchForm'
})(Form);

export default SearchForm;
