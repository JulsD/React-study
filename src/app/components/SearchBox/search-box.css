.root {
  width: 100%;
  padding: 30px 0 15px;
}
.title {
  text-transform: uppercase;
  margin-bottom: 20px;
  font-weight: 600;
}
.searchInput {
  width: 100%;
  background-color: black;
  color: #fff;
  border: none;
  border-bottom: 1px solid #f40612;
  margin-bottom: 15px;
  outline: none;
  padding: 15px 20px;
}
.switcher {
  width: 50%;
  text-align: left;
  display: inline-block;
}
.switcherTitle {
  text-transform: uppercase;
  display: inline-block;
  vertical-align: middle;
  font-weight: 500;
}
.switcher label {
  display: inline-block;
  vertical-align: middle;
  margin-left: 10px;
  position: relative;
}
.switcher label input {
  visibility: hidden;
  position: absolute;
  z-index: -1;
}
.submit {
  width: 50%;
  text-align: right;
  display: inline-block;
}