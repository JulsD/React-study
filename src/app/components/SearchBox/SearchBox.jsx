import React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import SearchForm from './SearchForm';

class SearchBox extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      searchText: '',
      searchBy: 'title'
    };

    this.handleInputChange = this.handleInputChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleInputChange(event) {
    const target = event.target;
    const value = target.value;
    const name = target.name;

    this.setState({
      [name]: value
    });
  }

  handleSubmit(event) {
    event.preventDefault();
    this.props.history.push({
      pathname: '/search',
      search: this.props.form.searchBy + '=' + this.props.form.searchText
    })
  }

  render() {
    return (
        <SearchForm onSubmit={this.handleSubmit} onChange={this.handleInputChange}/>
    );
  }
}

function mapProps(state) {
  return {
    form: state.form
  }
}

export default withRouter(connect(mapProps)(SearchBox));