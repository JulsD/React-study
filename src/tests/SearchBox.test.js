import React from 'react';
import SearchBox from '../app/components/SearchBox';

test('SearchBox should render a <form>', () => {
  const wrapper = shallow(
    <SearchBox />
  );
  expect(wrapper).toMatchSnapshot();
});
