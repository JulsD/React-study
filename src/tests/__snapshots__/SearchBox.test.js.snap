// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox should render a <form> 1`] = `
<Connect(Form(withRouter(SearchBox)))
  destroyOnUnmount={true}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="search"
  getFormState={[Function]}
  initialValues={
    Object {
      "searchBy": "title",
      "searchText": "",
    }
  }
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  touchOnBlur={true}
  touchOnChange={false}
/>
`;
